charms/knative-eventing/requirements-unit.in

```diff
@@ -1,5 +1,23 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+# Please note this file introduces dependencies from the charm's requirements.in,
+# special attention must be taken when updating this or the other .in file to try
+# to avoid incompatibilities.
+# Rules for editing this file:
+#   * Removing a dependency that is no longer used in the unit test file(s)
+#     is allowed, and should not represent any risk.
+#   * Adding a dependency in this file means the dependency is directly used
+#     in the unit test files(s).
+#   * ALL python packages/libs used directly in the unit test file(s) must be
+#     listed here even if requirements.in is already adding them. This will
+#     add clarity to the dependency list.
```

Comment:
I'm not sure I agree with this.  This feels redundant makes things less clear rather than more.  I'm ok with the sentiment, but it is so easy to get wrong (because, when you do `import coverage` in the tests, nothing will warn you that the dependency works but isn't in `requirements-unit.txt`).  So since as a dev I can't rely on it, I rather it not be there at all

Reply:
If we do still agree we should do this, can we say "should be listed here" rather than "must be listed here".  "must" tells me "it will break if we dont", whereas something like "should be listed here even if it is already in requirements.in to make it clearer it is a dependency in the tests" or similar says its a convention we choose to follow.

Reply:
Resolving for now to merge, and will discuss separately and possibly revert after.